File: 1598/CrawlerLogFolder.java
/*
https://leetcode.com/problems/crawler-log-folder/description/?envType=daily-question&envId=2024-07-10
*/

class Solution {
    public int minOperations(String[] logs) {
        int currentDepthInTheDirectory=0, resultantMinOperationRequired=0, lengthOfLogs=logs.length;
        for(int i=0; i<lengthOfLogs; ++i){
            String currentLog=logs[i];
            int lengthOfCurrentLog=currentLog.length();
            if(currentLog.equals("./")) continue;
            else if(currentLog.charAt(lengthOfCurrentLog-1)=='/'&&
                    currentLog.charAt(lengthOfCurrentLog-2)=='.'&&
                    currentLog.charAt(lengthOfCurrentLog-3)=='.'){
                currentDepthInTheDirectory=Math.max(currentDepthInTheDirectory-1, 0);
            } else currentDepthInTheDirectory++;
        }
        while(currentDepthInTheDirectory>=1){
            resultantMinOperationRequired++;
            currentDepthInTheDirectory--;
        }
        return resultantMinOperationRequired;
    }
}

File: 1598/CrawlerLogFolder.cpp
/*
https://leetcode.com/problems/crawler-log-folder/description/?envType=daily-question&envId=2024-07-10
*/

class Solution {
public:
    int minOperations(vector<string>& logs) {
        int curdepth=0, res=0;
        for(auto &vals: logs){
            int sz=vals.size();
            if(vals=="./") continue;
            else if(vals[sz-1]=='/'&&vals[sz-2]=='.'&&vals[sz-3]=='.'){
                curdepth--;
                if(curdepth<0) curdepth=0;
            }
            else curdepth++;
        }
        while(curdepth>=1){
            res++, curdepth--;
        }
        return res;
    }
};